### PR TITLE
[US2361]Add unit test for spc watcher

### PR DESCRIPTION
**Pull request by sonasingh46**
This commit will do following:

1. Add unit tests for various components
   of SPC watcher.

2. Remove unwanted code.

3. Refactor code.

Signed-off-by: sonasingh46 <sonasingh46@gmail.com>

### CODE REVIEW

File: cmd/maya-apiserver/spc-watcher/controller.go [hunk 1]
```diff
@@ -63,6 +63,7 @@ type Controller struct {
 	// recorder is an event recorder for recording Event resources to the
 	// Kubernetes API.
 	recorder record.EventRecorder
+	queueLoad QueueLoad
```
Comment (princerachit) [inline cmd/maya-apiserver/spc-watcher/controller.go]: Comments about this field would help.

Comment (sonasingh46) [reply cmd/maya-apiserver/spc-watcher/controller.go]: Will update this as a part of next PR.

File: cmd/maya-apiserver/spc-watcher/handler.go [hunk 2]
```diff
@@ -39,10 +39,11 @@ func (c *Controller) syncHandler(key, operation string, object interface{}) erro
 	// Check if the event is for delete and use the spc object that was pushed in the queue
 	// for utilising details from it e.g. delete cas template name for storagepool deletion.
 	if operation == deleteEvent {
+		object := object.(*apis.StoragePoolClaim)
 		if object == nil {
-			glog.Error("Nil storagepoolclaim object found for storage pool deletion")
+			return fmt.Errorf("storagepoolclaim object not found for storage pool deletion")
 		}
-		spcGot = object.(*apis.StoragePoolClaim)
+		spcGot = object
```
Comment (avishnu) [inline cmd/maya-apiserver/spc-watcher/handler.go]: Why did we have to change this typecast logic? If this needs to be fixed, please take it up in the next commit.

Comment (sonasingh46) [reply cmd/maya-apiserver/spc-watcher/handler.go]: I will add this as a part of next PR

**Pull request merged**